internal/gps/checks.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import (
+	"github.com/golang/dep/internal/gps/pkgtree"
+)
+
+func FindIneffectualConstraints(manifest Manifest, packageTree pkgtree.PackageTree) []ProjectRoot {
```

Comment:
given the breadth of things this func is checking for, maybe a better name would be `FindIneffectualRules`